script.js

```diff
@@ -1,19 +1,99 @@
-const filterSection = document.querySelector(".filter");
-const filterButton = document.querySelector("#filterButton");
-const filterCloseButton = document.querySelector(".filterCloseButton");
-
+// -------------------- MOBILE MENU --------------------
 document.querySelector(".main-nav-toggle").addEventListener("click", () => {
   document.querySelector(".main-nav").classList.toggle("open");
 });
 
-filterButton.addEventListener("click", () => {
-  filterSection.style.display = "flex";
-  filterButton.style.display = "none";
-  filterButton.setAttribute("aria-expanded", true);
-});
+// -------------------- FILTER SECTION  --------------------
+// Only runs filter code if it is on the right html file
+const host = "http://127.0.0.1:5500/";
+if (window.location.href == host + "challenges.html") {
+  const filterSection = document.querySelector(".filter");
+  const filterButton = document.querySelector("#filterButton");
+  const filterCloseButton = document.querySelector(".filterCloseButton");
+
+  filterButton.addEventListener("click", () => {
+    filterSection.style.display = "flex";
+    filterButton.style.display = "none";
+    filterButton.setAttribute("aria-expanded", true);
+  });
+
+  filterCloseButton.addEventListener("click", () => {
+    filterSection.style.display = "none";
+    filterButton.style.display = "block";
+    filterButton.setAttribute("aria-expanded", false);
+  });
+}
+
+// -------------------- MODAL --------------------
+
+// Trigger "book this room" to open modal
+const openModal = document.querySelectorAll(".modal-open");
```

Comment:
When I saw `openModal` elsewhere in the code, I was confused because it was used as an array even though the name did not suggest that the value would be an array. Conclusion: the naming could be better. Usually, arrays will be named in a way that clearly indicates plurality, and the current name not only implies a singular value, it implies that the value would be a modal. I would call this variable something along the lines of `modalOpenButtons`.